docs: add template for examples (#497)
jina-ai/examples: example_template.md

> <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

Reviewer: Opinion: Let's add the title above the table of contents!

Reviewer 2: Opinion: Lets remove the table of contents

Reviewer 3: I'm not sure about removing it, I think it's a clear way to show what will be covered

**Table of Contents**
- [Run the EXAMPLE NAME](#run-the-example-name)
  - [🗝️ Expected knowledge](#Expected-knowledges)
  - [🔮 Overview of the files](#Overview-of-the-files)
  - [🏃 Run the Flows](#run-the-flows)
  - [🌀 Flow Diagram](#flow-diagram)
  - [🌟 Results](#results)
  - [🧞 OPTIONAL: EXTRA KNOWLEDGE FROM THIS EXAMPLE](#optional)
  - [💫 Deploy with Docker](#Deploy-with-docker)
  - [Next steps](#next-steps)
  - [Community](#community)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

# Run the EXAMPLE NAME

Write a short description of
1. What is this?
2. What is the expected outcome?
3. What the user will learn?

> If you want to run the example with Docker, check the instruction at the end of this README.  

Reviewer 2: ```suggestion
If you want to run this example with Docker, check the instruction at the end of this README.  
```

Reviewer 3: I tried to avoid repeat "this" twice in the same sentence.

## 🗝️ Expected knowledge

> Write a list of whatever the user is expected to know.

Reviewer 2: ```suggestion
Outline in bullet points anything the user is expected to know before diving in. For example:
```

> 1. Read up on [Jina 101](http://101.jina.ai).

Reviewer: Opinion: I don't think we should expect the user to have read 101 before they see examples. This will confuse them to much, the example are to show the power of Jina to users.

Reviewer 2: second this

Reviewer 3: Fair enough, I'll remove it

> 2. Python knowledge.

Reviewer 2: This is very ambiguous

> ### Install requirements

Reviewer 2: Add a section before this for:
1. cloning the repo and cd into directory
2. Dataset download

Reviewer: Second this

> There are some requirements to run this example, to download everything you need, just run in your terminal:

Reviewer 2: ```suggestion
In order to install all the Python libraries required you can run the following in your terminal:
```

```
pip install -r requirements.txt
```

## 🔮 Overview of the files

Add a list with all files in the example:

> * index.yml: YAML file for indexing

Reviewer 2: ```suggestion
* `index.yml:` YAML file for indexing
```

* query.yml: YAML file for querying
* encoder.yml: YAML file for encoder pod
* /workspace: Directory that stores indexed files (embeddings and documents). Automatically created after the first indexing.

> ## 🏃 Run the Flows

Reviewer: I would "step" to each of these titles. 

Step 1. Run the Index Flow 
Step 2. Run the Query Flow 

https://developers.google.com/tech-writing/one/lists-and-tables

> We usually need at least two Flows in Jina. One for Indexing and one for Querying.

Reviewer 2: ```suggestion
Most Jina applications will use two Flows; One for Indexing and one for Querying.
```

Reviewer 3: Sounds better, I'll change it. About the ";" it's discouraged to use those according to the Google workshop we had, so I think I should stay with normal period?

### Index Flow

Describe the Index Flow. Be as specific as possible. You are encouraged to user code snippets, images, or whatever helps to clarify.

### Query Flow

Describe the Query Flow. Be as specific as possible. You are encouraged to user code snippets, images, or whatever helps to clarify.

## 🌀 Flow diagram

Show the Flow for this example.

## 🌟 Results

> Short description of the results and how to interpret them if needed.

Reviewer 2: If this is optional we should change the title to indicate that

Reviewer 3: It's optional if the results need to be interpreted. Some times the highest in the list is what you would expect or things like that. But the Results itself shouldn't be optional

> ## 🧞 OPTIONAL: EXTRA KNOWLEDGE FROM THIS EXAMPLEs

Reviewer: Cap locks ;) People will copy and paste this without noticing.

Reviewer 3: I had the caps so they would notice and change to something specific to them, but Agnes also mentioned this so I better change It

> ## 🧞 OPTIONAL: EXTRA KNOWLEDGE FROM THIS EXAMPLEs

Reviewer 2: ```suggestion
## 🧞 Optional: Read more about topics mentioned in this example
```

Reviewer 3: Whatever was in caps was supposed to be modified to the specific use case of each example

You can use this section if you feel extra knowledge could be useful.
QueryLanguage for example.

## 💫 Deploy with Docker

Add instructions on how to run with Docker:

> For a one line deployment, check out docker instructions here.

Reviewer 2: Where is here?

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

# Run the EXAMPLE NAME

Write a short description of
1. What is this?
2. What is the expected outcome?
3. What the user will learn?

If you want to run the example with Docker, check the instruction at the end of this README.  

## 🗝️ Pre requirements

Outline in bullet points anything the user is expected to have before diving in. For example:

1. You have a working Python 3.8 environment. 
2. We recommend creating a [new python virtual envoriment](https://docs.python.org/3/tutorial/venv.html) to have a clean install of Jina and prevent dependency clashing.   
3. You have at least 8GB of free space on your hard drive. 

### Install requirements

In order to install all the Python libraries required you can run the following in your terminal:

```
pip install -r requirements.txt
```

## 🔮 Overview of the files

Add a list with all files in the example:

* `index.yml`: YAML file for indexing
* `query.yml`: YAML file for querying
* `encoder.yml`: YAML file for encoder pod
* `/workspace: Directory that stores indexed files (embeddings and documents). Automatically created after the first indexing.

> ## 🏃 Run the Flows

Reviewer: These instructions explain how to build the example yourself and deploy it with Python. If you want to skip the building skips and just run the app, check out the Docker section below.

Reviewer 3: I added it also. There was another comment from Agnes I didn't see before so it's all now in the latest PR

  - [🏃 Run the Flows](#run-the-flows)
  - [🌀 Flow Diagram](#flow-diagram)
  - [🌟 Results](#results)
  - [🧞 OPTIONAL: EXTRA KNOWLEDGE FROM THIS EXAMPLE](#optional)
  - [💫 Deploy with Docker](#Deploy-with-docker)
  - [Next steps](#next-steps)
  - [Community](#community)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

# Run the EXAMPLE NAME

Write a short description of
1. What is this?
2. What is the expected outcome?
3. What the user will learn?

If you want to run the example with Docker, check the instruction at the end of this README.  

## 🗝️ Pre requirements

Outline in bullet points anything the user is expected to have before diving in. For example:

1. You have a working Python 3.8 environment. 
2. We recommend creating a [new python virtual envoriment](https://docs.python.org/3/tutorial/venv.html) to have a clean install of Jina and prevent dependency clashing.   
3. You have at least 8GB of free space on your hard drive. 

### Install requirements

Begin by cloning the repo so you can get the required files and datasets:

```sh
git clone https://github.com/jina-ai/examples
````

And enter the correct folder:

```sh

> cd examples/example_to_use

Reviewer 4: we need to tell user that change `example_to_use` to the corresponding name of the folder? To avoid someone directly copy-paste and run this command

Reviewer 3: This is a template for the devs who will write the documentation not really the users

Reviewer 4: Oh ok, great!

```

> On your terminal, you should now be located in you the wikipedia-sentences folder. Let's install Jina and the other required python libraries. For futher information on installing Jina check out [documentation](https://docs.jina.ai/chapters/core/setup/).

Reviewer 4: why it is `wikipedia-sentences folder`, just for an example?

Reviewer 3: Yep, this is a template only, so it's supposed to be used as a base to write something else.
